src/components/screens/AvailableVehicles.tsx

```diff
@@ -36,13 +38,15 @@ export const AvailableVehicles = ({
         }
       } catch (error) {
         console.error('There was a problem with the fetch operation:', error);
+      } finally {
+        setLoading(false);
       }
     };
 
     fetchData();
   }, []);
 
-  if (!availableVehiclesData?.length) {
+  if (loading) {
```

Comment:
the headline could be already there on loading to see what is loading. like on home screen sections as well, right!?

Reply:
i did it in release branch already